src/app/business/common/sagas/index.js

```diff
@@ -9,13 +9,51 @@ import cookie from 'cookie-parse';
 import {fetchRefresh} from '../../user/api';
 import {refresh as refreshActions, signOut} from '../../user/actions';
 
+export const getJWTFromCookie = () => {
+    let jwt;
+    if (typeof window !== 'undefined') {
+        const cookies = cookie.parse(window.document.cookie);
+        if (cookies['header.payload']) {
+            jwt = cookies['header.payload'];
+        }
+    }
+    return jwt;
+};
+
+
+export const tryRefreshToken = function* tryRefreshToken(action_error) {
+    // try to refresh token
+    const {res, error} = yield call(fetchRefresh);
+
+    // refresh token does not exist
+    if (error) { // redirect to login page
+        yield put(refreshActions.failure(error));
+        yield put(action_error(error));
+        yield put(signOut.success());
+    }
+    else {
+        yield put(refreshActions.success(res));
+
+        const jwt = getJWTFromCookie();
+
+        if (!jwt) {
+            yield put(action_error(error));
+            yield put(signOut.success());
+        }
+
+        return jwt;
+    }
+};
 
 export const fetchListSaga = (actions, fetchListApi) => function* fetchList({payload}) {
     const {error, status, list} = yield call(fetchListApi, payload);
 
     if (error) {
         console.error(error, status);
         yield put(actions.list.failure(error));
+        if (status === 401) {
+            yield tryRefreshToken(actions.list.failure);
```

Comment:
I feel we should do a `yield put(signOut.success());` here instead of trying to refresh the token. I see 2 reasons for this:
- if we got here, it's because we have a token but it is broken for some reason, so trying to refresh it will only cause us to be rejected for sure (1 extra call unnecessary) and we'll dispatch the `action.list.failure` action twice (was dispatched just above)
- and in the unlikely case the tryRefreshToken is successful, we'll be in a weird state where the action we just tried failed, isn't retried and leaves us wondering why nothing really happened in the UI.

So for me I see two possibilities either use `signOut.success` or implement a proper retry strategy but this requires a bit more code.

Reply:
I agree, will do the change